src/renderer/api/navidrome/navidrome-api.ts

```diff
@@ -182,54 +228,78 @@ axiosClient.interceptors.response.use(
       }
     }
 
+    authSuccess = true;
+
     return response;
   },
   (error) => {
     if (error.response && error.response.status === 401) {
-      toast.error({
-        message: 'Your session has expired.',
-      });
-
       const currentServer = useAuthStore.getState().currentServer;
 
-      if (currentServer) {
-        const serverId = currentServer.id;
-        const token = currentServer.ndCredential;
-        console.log(`token is expired: ${token}`);
-        useAuthStore.getState().actions.updateServer(serverId, { ndCredential: undefined });
-        useAuthStore.getState().actions.setCurrentServer(null);
-      }
-    }
+      if (localSettings && currentServer?.savePassword) {
+        // eslint-disable-next-line promise/no-promise-in-callback
+        return localSettings
+          .passwordGet(currentServer.id)
+          .then(async (password: string | null) => {
+            authSuccess = false;
 
-    return Promise.reject(error);
-  },
-);
+            if (password === null) {
+              throw error;
+            }
 
-const parsePath = (fullPath: string) => {
-  const [path, params] = fullPath.split('?');
+            if (shouldDelay) {
+              await waitForResult();
 
-  const parsedParams = qs.parse(params);
+              // Hopefully the delay was sufficient for authentication.
+              // Otherwise, it will require manual intervention
+              if (authSuccess) {
+                return axiosClient.request(error.config);
+              }
 
-  // Convert indexed object to array
-  const newParams: Record<string, any> = {};
-  Object.keys(parsedParams).forEach((key) => {
-    const isIndexedArrayObject =
-      typeof parsedParams[key] === 'object' && Object.keys(parsedParams[key] || {}).includes('0');
+              throw error;
+            }
 
-    if (!isIndexedArrayObject) {
-      newParams[key] = parsedParams[key];
-    } else {
-      newParams[key] = Object.values(parsedParams[key] || {});
-    }
-  });
+            shouldDelay = true;
 
-  const notNilParams = omitBy(newParams, (value) => value === 'undefined' || value === 'null');
+            // Do not use axiosClient. Instead, manually make a post
+            const res = await axios.post(`${currentServer.url}/auth/login`, {
+              password,
+              username: currentServer.username,
+            });
 
-  return {
-    params: notNilParams,
-    path,
-  };
-};
+            if (res.status !== 200) {
+              throw new Error('Failed to authenticate');
+            }
```

Comment:
An additional conditional could be added to catch the `http 429` error so that it doesn't log you out in the case that multiple login requests are sent.

Reply:
Given that the default timeout is 20 seconds, I think it probably makes more sense to just give them a warning